x-pack/plugin/rest-compatibility/build.gradle

```diff
@@ -18,6 +18,7 @@
  */
 
 apply plugin: 'elasticsearch.esplugin'
+apply plugin: 'elasticsearch.rest-test'
```

Comment:
Can you change this `apply plugin: 'elasticsearch.java-rest-test'` and move the test from `src/test` to `src/javaRestTest` (once you refresh the Grade project src/javaRestTest will show up as a normal source folder)

This helps to disambiguate which kind of REST test allowing for different configurations if needed.

Reply:
sure, good idea.
btw I have few questions. 
I figured out I need to set `  setting 'xpack.license.self_generated.type', 'trial'` in testClusters configuration in order to javaRestTest task to run? (otherwise it fails complaining about `Missing plugin [x-pack-core]`)
should I put that in `rest-compatibility/src/javaRestTest` ? I noticed a lot of modules are using `module-name/qa/javaRestTest` I couldn't find a single module in x-pack not using that convention. Should I follow it as well?

Reply:
1. That's weird since it is in x-pack and shouldn't need to specify the license. It is probably more likely that the new plugin needs (if not already there)
```
esplugin {
  extendedPlugins = ['x-pack-core']
```
or possibly
```
dependencies {
  javaRestTestImplementation project(path: xpackModule('core'), configuration: 'testArtifacts')
}
```
or possibly 

```
dependencies {
  javaRestTestImplementation project(path: xpackModule('core'))
}
```
Fine to check it in as-is, I can take a look afterwards.

2. That is generally the preferred pattern, but either should work and we can move things around as needed later too.